### PR TITLE
Vn 157 create lambda that reads from s3 bucket

### PR DESCRIPTION
**Pull request by SegevHaviv**

### CODE REVIEW

File: src/lambdas/macforceIntegrationObjectsSync/storageHelper/storageHelper.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  const newObjectKey = get(filteredNewObjectsLists, '[0].Key');
+  const oldObjectKey = get(filteredOldObjectsLists, '[0].Key');
+
+  return Promise.all([
+    isOldDataExist ? storageManager.readFromMacforceBucket(oldObjectKey) : Promise.resolve(undefined),
+    storageManager.readFromMacforceBucket(newObjectKey)
+  ]);
+};
+
+// TODO SET PATHS CORRECTLY BY CONFIG
```
Comment (VinoLora) [inline src/lambdas/macforceIntegrationObjectsSync/storageHelper/storageHelper.ts:38]: Don't forget to remove or to do the 'TODO'

File: src/lambdas/macforceIntegrationObjectsSync/macforceIntegrationObjectsSync.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    if (!eventsParamsToExecute || !eventsParamsToExecute?.length) {
+      console.info('No new events detected.');
+      return;
+    }
+
+    // @ts-ignore
+    const combinedEvents = combineUpdateEvents(eventsParamsToExecute);
+
+    const sendEventResponse = await sendEvents(entityName, combinedEvents);
+    if (sendEventResponse.some(singleResponse => !singleResponse || (singleResponse && singleResponse.FailedEntryCount))) {
```
Comment (VinoLora) [inline src/lambdas/macforceIntegrationObjectsSync/macforceIntegrationObjectsSync.ts:42]: You can extract this condition to function or const  like 'hasSendFailures' for better readability

File: src/lambdas/macforceIntegrationObjectsSync/eventsHelper/eventsHelper.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import { PutEventsRequestEntryList } from 'aws-sdk/clients/eventbridge';
+import { EntityNames, EventParams } from '../types';
+import { entityNameToAppflowName } from '../consts';
+
+const config = require('@venncity/nested-config')(__dirname);
+
+const eventBridgeBusName = config.get('eventbridge.eventBusName');
+const eventBridge = new AWS.EventBridge();
+
+// const eventBridgeBusName = `macforce-event-test`; // todo should eventbridge be based on stage
```
Comment (VinoLora) [inline src/lambdas/macforceIntegrationObjectsSync/eventsHelper/eventsHelper.ts:13]: Don't forget to remove commented out code

File: src/lambdas/macforceIntegrationObjectsSync/eventsHelper/eventsHelper-test.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+describe('eventsHelper', () => {
+  afterEach(jest.clearAllMocks);
+  afterAll(jest.resetAllMocks);
+
+  describe('sendSingleEvent', () => {
+    test('Should send event with the correct given params & enrich it correctly', async () => {
+      const event = generatePropertyEventParams();
+
+      const response = await sendSingleEvent(EntityNames.Property, [event as any]);
+      console.log('response', response);
```
Comment (VinoLora) [inline src/lambdas/macforceIntegrationObjectsSync/eventsHelper/eventsHelper-test.ts:31]: You left the log here on purpose or by mistake?

**Review by VinoLora — APPROVED** [review]
Impressive!
Just don't forget to clean the code from todos's and commented out code leftovers

File: src/lambdas/macforceIntegrationObjectsSync/storageHelper/storageHelper.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    console.error(`Could not find any new data for ${entityName}`);
+    return [undefined, undefined];
+  }
+
+  const isOldDataExist = Boolean(oldObjectsListsForEntity?.Contents?.length);
+  if (!isOldDataExist) {
+    console.warn('No old data found, no diff with old data will be done');
+  }
+
+  // check if i should iterate between all the fetched objects or the first one is sufficient
```
[the remark this answers is not in the document]
Comment (SegevHaviv) [reply src/lambdas/macforceIntegrationObjectsSync/storageHelper/storageHelper.ts:28]: It's a leftover from the coding process, I'll remove it.

File: src/lambdas/macforceIntegrationObjectsSync/eventsHelper/eventsHelper-test.ts (hunk 4 above)
Comment (SegevHaviv) [reply src/lambdas/macforceIntegrationObjectsSync/eventsHelper/eventsHelper-test.ts]: Mistake, removed.

**Pull request merged**